src/Data/Lens/Index.purs

```diff
@@ -27,14 +27,16 @@ import Data.Traversable (traverse)
 -- | set (ix 1) 8888 [0, 1, 2] == [0,8888,2]
 -- | ```
 -- |
--- | Note the use of `preview` rather `view`.
+-- | Note the use of `preview` rather `view`. That's because the optic is 
+-- | a `Data.Lens.Traversal` tailored to the case where there's a single element
+-- | of interest.
```

Comment:
One last nitpick :smile::

I'm not sure about this explanation - `preview` is because a `Traversal` might "fail" when being read from (in the sense that it can be pointing at nothing, in which case `preview`ing it gives you `Nothing`). `view` only works for `Lens`, `Iso`, things that are guaranteed to have a value.

Reply:
The problem is that `Index` and `At` are closely linked by usage. From the point of view of a person choosing between them, they *both* can return a `Nothing`. That the mechanism is different is not the sort of detail we should force users to learn, at least not in the quick reference.

Reply:
If we did, we'd just be in the situation produced by the only block comment in the V6 Unix kernel (which I read with my own eyes). It was the the code that did process switching. The comment was about six lines long. The last sentence was "You are not expected to understand this."

That was not the place for the discussion. The place was [Lion's book](https://en.wikipedia.org/wiki/Lions'_Commentary_on_UNIX_6th_Edition%2C_with_Source_Code).

Reply:
Well, whad'ya know: https://en.wikipedia.org/wiki/Lions'_Commentary_on_UNIX_6th_Edition%2C_with_Source_Code#"You_are_not_expected_to_understand_this"

Reply:
Hmm, although I think maybe I see what you were getting at with

> tailored to the case where there's a single element of interest.

When I first read it I wasn't sure what was going on here, as it still sounded like an explanation of `preview`, but it's not related to that at all - so perhaps just breaking the sentence up would work too; "preview-because-traversal. traversal-is-affine". 

I'm also fine with not explaining the exact nature of the traversal though, as it the 0-or-1 nature of it seems to follow "obviously" (?) from the purpose of the class.

Reply:
I think you're saying you'll accept the pull request...?